### PR TITLE
feat(var) load default var indexes

### PR DESCRIPTION
**Pull request by chronolaw**
Add a new directive lua_kong_load_default_var_indexes, which can simplify nginx.conf.

### CODE REVIEW

**Review by fffonion — APPROVED** [review]

File: src/ngx_http_lua_kong_var.c [hunk 1]
```diff
@@ -18,6 +18,101 @@
...
+/* default variable indexes will be loaded */
+static ngx_str_t default_vars[] = {
+    ngx_string("args"),
+    ngx_string("is_args"),
+    ngx_string("bytes_sent"),
+    ngx_string("content_type"),
+    /* ngx_string("host"), */
+
+/* http request headers */
+#if 1
```
Comment (dndx) [inline src/ngx_http_lua_kong_var.c:30]: We should remove this.

File: src/ngx_http_lua_kong_var.c [hunk 2]
```diff
@@ -18,6 +18,101 @@
...
+
+
+static char *
+ngx_http_lua_kong_load_default_var_indexes(ngx_conf_t *cf)
+{
+    ngx_str_t                     *var;
+    ngx_int_t                      index;
+
+    for (var = default_vars; var->len; var++) {
+
```
Comment (dndx) [inline src/ngx_http_lua_kong_var.c:101]:
```suggestion
```

File: t/006-default_indexed-var.t [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+--- response_body
+ok
+--- no_error_log
+[error]
+[crit]
+[alert]
+
+
+
+=== TEST 2: variable $is_args$args
```
Comment (dndx) [inline t/006-default_indexed-var.t:43]:
```suggestion
=== TEST 2: variable $is_args, $args
```

File: t/006-default_indexed-var.t [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+
+__DATA__
+
+=== TEST 1: sanity: directive works well
+--- http_config
+    lua_kong_load_var_index default;
+
+--- config
+    location = /test {
+        content_by_lua '
```
Comment (dndx) [inline t/006-default_indexed-var.t:28]: Let's change all to use `content_by_lua_block` in the tests. We should not be using `content_by_lua` for new test files as it is no longer recommended to use by OpenResty.

File: t/006-default_indexed-var.t [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+get variable value '?' by index
+get variable value 'hello=world' by index
+--- no_error_log
+[error]
+[crit]
+[alert]
+
+
+
+=== TEST 3: variable $scheme$host$request_uri
```
Comment (dndx) [inline t/006-default_indexed-var.t:71]:
```suggestion
=== TEST 3: variable $scheme, $host, $request_uri
```

File: t/006-default_indexed-var.t [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+--- response_body
+http2:
+--- error_log
+get variable value '' by index
+--- no_error_log
+[error]
+[crit]
+[alert]
+
+=== TEST 12: variable $content_type$bytes_sent
```
Comment (dndx) [inline t/006-default_indexed-var.t:386]:
```suggestion
=== TEST 12: variable $content_type, $bytes_sent
```

File: src/ngx_http_lua_kong_var.c (hunk 2 above)
Comment (chronolaw) [reply src/ngx_http_lua_kong_var.c:101]: done

File: src/ngx_http_lua_kong_var.c (hunk 1 above)
Comment (chronolaw) [reply src/ngx_http_lua_kong_var.c:30]: done

File: t/006-default_indexed-var.t (hunk 3 above)
Comment (chronolaw) [reply t/006-default_indexed-var.t:43]: done

File: t/006-default_indexed-var.t (hunk 5 above)
Comment (chronolaw) [reply t/006-default_indexed-var.t:71]: done

File: t/006-default_indexed-var.t (hunk 6 above)
Comment (chronolaw) [reply t/006-default_indexed-var.t:386]: done

File: t/006-default_indexed-var.t (hunk 4 above)
Comment (chronolaw) [reply t/006-default_indexed-var.t:28]: done

**Review by dndx — APPROVED** [review]

**Pull request merged**